_posts/2022-01-10-reassign-partitions.md

```diff
@@ -0,0 +1,539 @@
+          {
+            "partition": "my-topic-two-0",
+            "size": 0,
+            "offsetLag": 0,
+            "isFuture": false
+          }
+        ]
+      }
+    ]
+  } // ...
+```
+
+Now we can try to move `my-topic-two-1` to use the log directory `/var/lib/kafka/data-1/kafka-log0` on broker 0.
+
+
+We can again generate the `reassignment.json` in the same way we generated it for the other examples.
+
+```json
+[
+  {
+    "topic": "my-topic-two",
+    "partition": 0,
+    "replicas": [
+      0,
+      3,
+      1,
+      2
+    ],
+    "log_dirs": [
+      "any",
+      "any",
+      "any",
+      "any"
+    ]
+  },
+  {
+    "topic": "my-topic-two",
+    "partition": 1,
+    "replicas": [
+      1,
+      0,
+      2,
+      3
+    ],
+    "log_dirs": [
+      "any",
+      "any",
+      "any",
+      "any"
+    ]
+  },
+  {
+    "topic": "my-topic-two",
+    "partition": 2,
+    "replicas": [
+      2,
+      1,
+      3,
+      0
+    ],
+    "log_dirs": [
+      "any",
+      "any",
+      "any",
+      "any"
+    ]
+  }
+]
+```
+
+Let's edit our `reassignment.json`and make the changes in log directory for partition 1.
+
+```json
+[
+  {
+    "topic": "my-topic-two",
+    "partition": 0,
+    "replicas": [
+      0,
+      3,
+      1,
+      2
+    ],
+    "log_dirs": [
+      "any",
+      "any",
+      "any",
+      "any"
+    ]
+  },
+  {
+    "topic": "my-topic-two",
+    "partition": 1,
+    "replicas": [
+      1,
+      0,
+      2,
+      3
+    ],
+    "log_dirs": [
+      "any",
+      "/var/lib/kafka/data-1/kafka-log0",
+      "any",
+      "any"
+    ]
+  },
+  {
+    "topic": "my-topic-two",
+    "partition": 2,
+    "replicas": [
+      2,
+      1,
+      3,
+      0
+    ],
+    "log_dirs": [
+      "any",
+      "any",
+      "any",
+      "any"
+    ]
+  }
+]
+```
+Now we can use the `--execute` mode of the tool to apply our `reassignment.json` file and after that use the `--verify` mode to check if the assignment is done or not just like we did in the above steps.
+
+ Once the assignment is done, you can check whether the log directory for partition 1 is now using the log directory `/var/lib/kafka/data-1/kafka-log0`
+
+```json
+[
+  {
+    "broker": 0,
+    "logDirs": [
+      {
+        "logDir": "/var/lib/kafka/data-0/kafka-log0",
+        "error": null,
+        "partitions": [
+          {
+            "partition": "my-topic-two-2",
+            "size": 0,
+            "offsetLag": 0,
+            "isFuture": false
+          }
+        ]
+      },
+      {
+        "logDir": "/var/lib/kafka/data-1/kafka-log0",
+        "error": null,
+        "partitions": [
+          {
+            "partition": "my-topic-two-1",
+            "size": 0,
+            "offsetLag": 0,
+            "isFuture": false
+          },
+          {
+            "partition": "my-topic-two-0",
+            "size": 0,
+            "offsetLag": 0,
+            "isFuture": false
+          }
+        ]
+      } // ...
```

Comment:
I think you should complete it to make it a valid JSON.

```suggestion
      }
    }
  }
]
```

Reply:
Actually it was a big `JSON` so I put in the part  where we are making the changes. Do we need to put the whole `json` ?

Reply:
I think you make it much more readable for users if you put in a valid piece of the JSON => so for example in this case, you can also maybe remove the starting braces instead of adding the closing one
```
[
  {
    {
```
part as well.